### PR TITLE
Convert this project to use Nix flakes

### PR DESCRIPTION
**Pull request by notgne2**

### CODE REVIEW

**Review by zhenyavinogradov — CHANGES_REQUESTED** [review]
There are errors in CI, pipeline steps need to be updated as well. Other than that, looks good

**Comment by notgne2** [comment]
https://hydra.nixos.org/build/129701778 😠

**Comment by Lucus16** [comment]
I'm okay with adding support for flakes but I want nixfmt to continue to be usable without using flakes so please don't remove or significantly change the existing Nix files.

**Comment by SuperSandro2000** [comment]
> to continue to be usable without using flakes

This is what flake-compat does. You don't need to have flakes enabled for it.

File: flake.nix [hunk 1]
```diff
@@ -0,0 +1,93 @@
+# © 2019 Serokell <hi@serokell.io>
+# © 2019 Lars Jellema <lars.jellema@gmail.com>
+#
+# SPDX-License-Identifier: MPL-2.0
+
+{
+  description = "A Simple multi-profile Nix-flake deploy tool.";
```
Comment (0x4A6F) [inline flake.nix:7]: Looks like copypasta from deploy-rs, could use a better description:  A formatter for Nix code

**Pull request merged**